lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -0,0 +1,285 @@
+        $index = str_replace('field_', '', $index);
+
+        $data[$index]['fact'] = dosomething_helpers_issetor($fact_data['fact']);
+        $data[$index]['id'] = dosomething_helpers_issetor($fact_data['nid']);
+        $data[$index]['footnote'] = (int) dosomething_helpers_issetor($fact_data['footnotes']);
+        $data[$index]['source'] = dosomething_helpers_issetor($fact_data['sources']);
+
+      }
+
+      $data['sources'] = dosomething_helpers_issetor($fact_vars['sources']);
+
+      return $data;
+    }
+
+    return NULL;
+
+  }
+
+
+  protected function getIssue() {
+    // @TODO: Need to find out if this is allowed to be a field with multiple values...?
+    $issue_id = dosomething_helpers_extract_field_data($this->node->field_issue);
+
+    if ($issue_id) {
+      $issue = $this->getTaxonomyTerm($issue_id);
+
+      return $issue;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getPrimaryCause() {
+    $cause_id = dosomething_helpers_extract_field_data($this->node->field_primary_cause);
+
+    if ($cause_id) {
+      $cause = $this->getTaxonomyTerm($cause_id);
+
+      return $cause;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTags() {
+    $data = array();
+
+    $tag_ids = dosomething_helpers_extract_field_data($this->node->field_tags);
+
+    if ($tag_ids) {
+      foreach ($tag_ids as $tid) {
+        $data[] = $this->getTaxonomyTerm($tid);
+      }
+
+      return $data;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTaxonomyTerm($tid) {
+    $data = array();
+
+    $taxonomy = taxonomy_term_load($tid);
+
+    $data['id'] = $taxonomy->tid;
+    $data['name'] = strtolower($taxonomy->name);
+
+    return $data;
+  }
+
+
+  protected function getTagline() {
+    return dosomething_helpers_extract_field_data($this->node->field_call_to_action);
+  }
+
+
+  /**
+   * @return string
+   */
+  protected function getTimeCommitment() {
+    // @TODO: I've renamed "active_hours" to "time_commitment" because it sounds more straightforward; but appreciate feedback.
+    return (float) dosomething_helpers_extract_field_data($this->node->field_active_hours);
+  }
+
+
+  protected function getScholarship() {
+    return dosomething_helpers_extract_field_data($this->node->field_scholarship_amount);
+  }
+
+
+  protected function getSolutionData() {
+    $data['copy'] = dosomething_helpers_extract_field_data($this->node->field_solution_copy);
+    $data['support_copy'] = dosomething_helpers_extract_field_data($this->node->field_solution_support);
+
+    return $data;
+  }
+
+
+  protected function getStaffPickStatus() {
+    return (bool) dosomething_helpers_extract_field_data($this->node->field_staff_pick);
+  }
+
+
+  protected function getStatus() {
+    return dosomething_helpers_extract_field_data($this->node->field_campaign_status);
+  }
+
+
+  protected function getType() {
+    return dosomething_helpers_extract_field_data($this->node->field_campaign_type);
+  }
+
+}
```

Comment:
The little NO sign is Github complaining that theres no empty line at the end of the file